### ISSUE TITLE
Feature request: xdg-screensaver support

### ISSUE THREAD

**Issue by orbea** [PCSX2/pcsx2#2079]
***PCSX2 version:***
`pcsx2-2017.09.17_af2278c_master-x86_64-1_git`

***Description of the issue:***

Would it be possible to have `xdg-screensaver` support on linux so that screensavers will not activate when using a gamepad? Common udev configurations will cause xorg to not recognize any input from gamepads and the screen will blank from inactivity while playing games which is rather annoying. Alternatively dbus can be used, but this could be unreliable when dbus upstream makes incompatible changes.

Workarounds includes using the program caffeine-ng or a script like this.
```
#!/bin/sh

while true; do
  xscreensaver-command -deactivate &
  sleep 270
done
```

***PC specifications:***
OS: `Slackware64-current`

**Comment by Hirato** [comment]
wasn't this fixed yonks ago with? 2b02a7433f5de72f15b66ae072796036422b3b74

**Comment by orbea** [comment]
While that commit does suggest that it has never worked for me with xscreensaver nor do I see any options to enable it. Am I overlooking something? It seems this is either not fully implemented or not entirely working.

**Comment by gregory38** [comment]
What is your current screensave timer ? Less than 2 minute ? It is working on my side (at least it worked on the past)

**Comment by orbea** [comment]
I have it set to 5 minutes. To test I set it to 1 minute with xscreensaver-demo and it the screen blanked after 1 minute.

**Comment by gregory38** [comment]
I think the above code might nit work with small timeout such as 1/2 minutes. You could try to replace `0xFFF` by `0xFF` to increase the frequency of screen saver reset.

**Comment by orbea** [comment]
I tried changing it to `0xFF` and the screen still blanked after 1 minute.
```
diff --git a/plugins/onepad/Linux/linux.cpp b/plugins/onepad/Linux/linux.cpp
index 2ec910fab..4b5e5d82d 100644
--- a/plugins/onepad/Linux/linux.cpp
+++ b/plugins/onepad/Linux/linux.cpp
@@ -105,7 +105,7 @@ PADupdate(int pad)
     // Emulate an user activity
     static int count = 0;
     count++;
-    if ((count & 0xFFF) == 0) {
+    if ((count & 0xFF) == 0) {
         // 1 call every 4096 Vsync is enough
         XResetScreenSaver(GSdsp);
     }
```

**Comment by gregory38** [comment]
Do you have a plain xorg or Xwayland ?

**Comment by orbea** [comment]
Plain xorg, I do not have any wayland here.
`xorg-server-1.19.4-x86_64-1`
`xscreensaver-5.37-x86_64-1`
This does work in other programs such as mpv, RetroArch and dolphin-emu so I think my set up does work.

**Comment by DIKH3D** [comment]
![image](https://user-images.githubusercontent.com/34287698/33634908-c2b8a568-da0d-11e7-9fe7-ac7aee9a7e44.png)

**Comment by orbea** [comment]
I think this is the source of the issue?

https://github.com/PCSX2/pcsx2/blob/a782e03ebf29055ea892ab5c98c37c0699153015/common/src/Utilities/Linux/LnxMisc.cpp#L58

Maybe it could just use the SDL2 function?

https://wiki.libsdl.org/SDL_DisableScreenSaver

**Comment by lheckemann** [comment]
So, I've dug into this. This should indeed work fine on X11, since the feature was fixed in e5f9923f76fcdbadc82143f3b16d30446eb42139. This doesn't use xdg-screensaver as suggested in the issue, but I'd say the issue could be closed, since this works correctly --- at least on X11.

This does leave the issue of non-X11 platforms open; in fact, PCSX2 currently crashes after 4096 frames on Wayland, because it still tries to reset the screensaver with the (nonexistent) X display! (PR to check this and fix the crash incoming).

A proper multiplatform approach would be nice to have; SDL's implementation would be nice, but is unsuitable because it's part of SDL's video subsystem, which PCSX2 doesn't use. Does anyone have ideas on how to solve this? xdg-screensaver would probably work, but is an awkward pile of shell and I'm not sure it's the best approach.
